docs/guide/custom-login-requirements.md

```diff
@@ -0,0 +1,250 @@
+* `afterPasswordVerified`: similar to `afterSubmit`, but requirements in this phase are displayed **after the password has been verified.** Best used for requirements that require special knowledge of the user, such as 2FA (Two-Factor Authentication).
+
+## Implementing `beforeSubmit` and `afterSubmit` requirements
+
+Here is server-side code for a simple requirement to solve a math problem when logging in. This example uses `phase: 'afterSubmit'`, because the requirement has a user interface and our recommendation is to let it appear on its own. One could also use `phase: 'beforeSubmit'` if displaying the math problem at the bottom of the login form is preferred, with no other code changes:
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    requirements(self) {
+      return {
+        add: {
+          MathProblem: {
+            phase: 'afterSubmit',
+            async props(req) {
+              if (!(req.session.mathProblem && req.session.mathAnswer)) {
+                const x = Math.ceil(Math.random() * 5);
+                const y = Math.ceil(Math.random() * 5);
+                req.session.mathProblem = `${x} + ${y}`;
+                req.session.mathAnswer = x + y;
+              }
+              return {
+                mathProblem: req.session.mathProblem
+              };
+            },
+            async verify(req) {
+              const answer = self.apos.launder.integer(req.body.requirements.MathProblem);
+              if ((!req.session.mathAnswer) || (answer !== req.session.mathAnswer)) {
+                throw self.apos.error('invalid', 'math problem incorrect');
+              }
+            }
+          }
+        }
+      };
+    }
+  };
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/login/index.js
+  </template>
+</AposCodeBlock>
+
+**What is happening here?**
+
+* At project level, custom login requirements must either be delivered as project-level configuration of the `@apostrophecms/login` module (in `modules/@apostrophecms/login` at project level). They may not be spread across several modules at project level, however you **may implement more than one** in the `requirements()` module initialization function of `@apostrophecms/login`. If you wish to ship a single custom login requirement as a public or private npm module for reuse, use the `improve: '@apostrophecms/login'` property so that your module is treated as an extra layer of configuration for `@apostrophecms/login`. Note that `improve` is not available at project level.
+* The `props()` function of the requirement is invoked before the form appears and returns an object whose properties become props for the custom Vue component shown below. Because it has access to `req`, this function can store information about the "challenge" presented by the requirement in `req.session`, so it remains available for verifying the result, **without disclosing the right answer** to the browser.
```

Comment:
What is `req`? I wouldn't expect someone jumping into this article to know that right away.